src/orders-v2/order-refunds/order-refunds-view-model.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  AppAvatarFragment,
+  OrderDetailsFragment,
+  OrderGrantedRefundFragment,
+  OrderGrantedRefundStatusEnum,
+  StaffMemberAvatarFragment,
+  TransactionActionEnum,
+  TransactionEventFragment,
+  TransactionEventTypeEnum,
+} from "@dashboard/graphql";
+
+export type OrderRefundDisplay = {
+  id: string;
+  type: "standard" | "manual";
+  status: OrderGrantedRefundStatusEnum;
+  amount: {
+    amount: number;
+    currency: string;
+  };
+  reason: string | null;
+  createdAt: string;
+  user: {
+    email: string;
+    firstName: string;
+    lastName: string;
+  } | null;
+};
+
+export type OrderRefundState =
+  | "noTransactionsToRefund"
+  | "allTransactionsNonRefundable"
+  | "refundable";
+
+export abstract class OrderRefundsViewModel {
+  static getRefundState(transactions: OrderDetailsFragment["transactions"]): OrderRefundState {
+    if (transactions.length === 0) {
+      return "noTransactionsToRefund";
+    }
+
+    if (
+      transactions.every(transaction => !transaction.actions.includes(TransactionActionEnum.REFUND))
+    ) {
+      return "allTransactionsNonRefundable";
+    }
```

Comment:
question: What about manual refunds? Previously we allowed creating new transactions with `refundedAmount` to mark cases just like that: when there are no transactions that can be refunded and store has refunded customer in some other way (e.g. wire transfer)

Reply:
logic is the same as it was before but instead of `true/false` I'm using state that combines boolean with reason. Previous logic:

```ts
export const canAddRefund = ({
  transactions,
  intl,
}: {
  transactions: OrderDetailsFragment["transactions"];
  intl: IntlShape;
}): CanAddRefund => {
  if (transactions.length === 0) {
    return {
      canRefund: false,
      reason: intl.formatMessage(refundGridMessages.noTransactionsToRefund),
    };
  }

  if (
    transactions.every(transaction => !transaction.actions.includes(TransactionActionEnum.REFUND))
  ) {
    return {
      canRefund: false,
      reason: intl.formatMessage(refundGridMessages.allTransactionsNonRefundable),
    };
  }

  return {
    canRefund: true,
  };
```